### DISCUSSION TITLE
Native Recaptch support for form builder plugin

### DISCUSSION THREAD

**Discussion by IzzyDevs** [payloadcms/payload#8257] [Feature Requests & Ideas]
Right now it's not apparent that there is any Recaptcha-like functionality to prevent automated form abuse, or I am just not familiar with the implementation... in which case, please clue me in.

I have an implementation of Google Recaptcha setup with the beta plugin-form-builder, using the npm patch-package at the moment for a template I am setting up. I just wanted to see what the general consensus is regarding this sort of functionality before attempting to contribute, so I don't waste anyone's time or move against the grain.

I was thinking we could just check for an environment variable to qualify whether or not to use the Recaptcha verification function server-side and in the onSubmit, or the payload config - so opting in is pretty simple. We can start with Google, and add others as we go.

Thoughts?

**Comment by GeorgeHulpoi** [comment]
https://github.com/GeorgeHulpoi/payload-recaptcha-v3

**Reply by IzzyDevs** [reply]
Hey George, thank you for the response
I wanted to try this one out, but I am having trouble getting it to work as expected.
I feel like I'm missing something fundamental about how this is meant to work.

I removed the patch, and installed the plugin you specified - but according to the documentation I need to add this to the collection config which would be `formSubmissions`.

`custom: {
		recaptcha: [
			{
				name: 'create',
				action: 'submit',
			},
			{
				name: 'update',
				action: 'modify',
			},
		],
	},`
	
...which is stored inside of the node_modules, which would have my writing patches again.

So I tried following the instructions [here](https://payloadcms.com/docs/beta/plugins/form-builder) for an override, I've made a few assumptions here:
- I don't need to specify a `slug`
- `custom` will be spread as a sibling to `fields`

`formSubmissionOverrides: {
        custom: {
          recaptcha: [
            {
              name: 'create',
              action: 'submit',
            },
            {
              name: 'update',
              action: 'modify',
            },
          ],
        },
      },`
      
 However, all my submissions are still being committed regardless of whether or not a pass the token.
